CHANGELOG.md

```diff
@@ -18,6 +18,9 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
   individually instead of being passed a directory by setting
   `merge-tools.$TOOL.diff-invocation-mode="file-by-file"` in config.toml.
 
+* When creating a new working-copy commit, the old commit is no longer
+  abandoned if it is still the working-copy commit of a different workspace.
+
```

Comment:
nit: I wouldn't add this changelog entry as the change isn't visible to user. (Or move it "fixed bugs" section if we think the current behavior is broken.)